src/pytest_mock_resources/fixture/database/redis.py

```diff
@@ -31,15 +44,25 @@ def create_redis_fixture(scope="function"):
     """
 
     @pytest.fixture(scope=scope)
-    def _(_redis_container, pmr_redis_config):
-        db = redis.Redis(host=pmr_redis_config.host, port=pmr_redis_config.port)
-        db.flushall()
+    def _(request, _redis_container, pmr_redis_config):
+        database_number = 0
+        if hasattr(request.config, 'workerinput'):
+            worker_input = request.config.workerinput
+            worker_id = worker_input['workerid']  # For example "gw0".
```

Comment:
Is it fair to assume that the `workerid` would always be "gw<some_number>"

Reply:
pytest's execnet hardcodes this naming scheme